cmd/collector/app/metrics.go

```diff
@@ -104,20 +109,27 @@ func newMetricsBySvc(factory metrics.Factory, category string) metricsBySvc {
 }
 
 func newCountsBySvc(factory metrics.Factory, category string, maxServiceNames int) countsBySvc {
+	// Add 3 to maxServiceNames threshold to compensate for extra slots taken by transport types
+	maxServiceNames = maxServiceNames + 3
```

Comment:
Not sure if this is a good idea. I'm combining otherServices with transport types so this makes the user-defined counter space smaller. So I'm compensating 3 here to make the test pass. The difference is that when user have `numOfServices > maxServiceNames` they might notice in metric the total number of serviceNames shows `maxServiceNames + 3`